transport_udp.go

```diff
@@ -59,7 +60,7 @@ type udpSender struct {
 
 // NewUDPTransport creates a reporter that submits spans to jaeger-agent.
 // TODO: (breaking change) move to transport/ package.
-func NewUDPTransport(hostPort string, maxPacketSize int) (Transport, error) {
+func NewUDPTransport(hostPort string, maxPacketSize int, logger log.Logger) (Transport, error) {
```

Comment:
This would be a breaking change. I would rather add another function like below and delegate to it from this one:
```
type UDPTransportParams struct {
    HostPort string
    MaxPacketSize int
    Logger log.Logger
}

func NewUDPTransportWithParams(params UDPTransportParams)
```

Reply:
Similarly for NewAgentClientUDP

Reply:
Made this change. I'm defaulting to log.StdLogger if no logger is passed (as is the case in NewUDPTransport and NewAgentClientUDP)